### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by zaibacu**
Python 3 got renamed `urlparse` module to `urllib.parse`, due to that, whole package does not work on Python 3.

### CODE REVIEW

**Comment by benbacardi** [comment]
Are there utilities in the six package we can use for these situations?

**Comment by zaibacu** [comment]
Yes, using `six` is better solution, fixed it

**Comment by mikebryant** [comment]
Looking good so far :)

If you take the ```matrix:
  allow_failures:
  - python: 3.5
    env: DJANGO_VERSION=">=1.8, < 1.9"
  - python: 3.5
    env: DJANGO_VERSION=">=1.9, < 1.10"``` chunk out of `.travis.yml` the pull request will reflect the results of the python3 tests.

File: nuit/utils.py [hunk 1]
```diff
@@ -13,10 +13,24 @@ def _closure(fn):
         return fn.__closure__
 
 
+def _code(fn):
```
Comment (mikebryant) [inline nuit/utils.py]: Could this be `six.get_function_code`?

File: nuit/utils.py [hunk 2]
```diff
@@ -13,10 +13,24 @@ def _closure(fn):
...
+        return fn.func_code
+    except AttributeError:
+        return fn.__code__
+
+
 def has_closure(fn):
     return hasattr(fn, 'func_closure') or hasattr(fn, '__closure__')
 
 
+def has_code(fn):
```
Comment (mikebryant) [inline nuit/utils.py]: Could `inspect.isfunction` be used instead here?

File: nuit/utils.py [hunk 3]
```diff
@@ -2,18 +2,33 @@
 
 from django.core.exceptions import PermissionDenied
 
+
+def _closure(fn):
```
Comment (mikebryant) [inline nuit/utils.py]: Could this be `six.get_function_closure`?

File: nuit/utils.py (hunk 1 above)
Comment (zaibacu) [inline nuit/utils.py]: Yes, I actually covered this case later, but forgot to push latest changes. Were investigating some weird issue related to '__func__'

File: nuit/utils.py (hunk 2 above)
Comment (zaibacu) [inline nuit/utils.py]: Does function code tag checking was actually trying to check if it is a function? I didn't looked at the code logic that deep, just done shallow refactoring to make it work

Comment (mikebryant) [inline nuit/utils.py]: I was just looking at https://docs.python.org/3/library/inspect.html#module-inspect, apparently `function` objects are the only things that have `__code__`

I'm not sure if there's anything else that might have these attributes.

The whole section here is about introspecting the view code we're linking to extract any logic about permissions, and run that without running the whole view. We're aiming to improve DRY by not needing to specify menu link visibility when you link to a view, as well as in the view itself.

Comment (zaibacu) [inline nuit/utils.py]: Ok, that does make sense. Any ideas on closure part?

Comment (mikebryant) [inline nuit/utils.py]: Not sure, can't find many references to how to tell if the attribute is supposed to be there or not. It's not listed on the inspect page.

**Comment by mikebryant** [comment]
The current error might be fixed by using `get_method_function()` from `six`, instead of `__func__`.
Partly a guess though, I haven't tested this

File: build-requirements.pip [hunk 4]
```diff
@@ -1,2 +1,3 @@
 gitversion
 beautifulsoup4
+six
```
Comment (mikebryant) [inline build-requirements.pip]: This shouldn't be necessary - `build-requirements.pip` isn't even used on Travis at the moment. (And I think I need to actually remove beautifulsoup4 too, not sure when that crept in.)

Comment (zaibacu) [inline build-requirements.pip]: My bad, commited wrong file :)

**Comment by mikebryant** [comment]
@zaibacu Could you have a look at #25 and confirm you're happy with the changes I made to your commits? :)

**Comment by mikebryant** [comment]
Closed in favour of #25

**Pull request closed without merging**